src/binary/Builtin.cc

```diff
@@ -632,6 +632,61 @@ extern "C" {
     return INT2FIX(strcoll(RSTRING_PTR(str1), RSTRING_PTR(str2)));
   }
 
+  // reads the value of a hash key
+  // used internally by stftime_wrapper
+  static int
+  tm_hash_value(VALUE hash, const char *key)
+  {
+    return NUM2INT(rb_hash_aref(hash, ID2SYM(rb_intern(key))));
+  }
+  
+  // converts a hash to a tm structure
+  // used internally by stftime_wrapper
+  void
+  hash_to_tm(VALUE hash, struct tm *res)
+  {
```

Comment:
According to this http://www.ex-parrot.com/~chris/random/initialise.html something like `struct tm timeinfo = {0};` would be more correct than `memset(&timeinfo, 0, sizeof timeinfo);`. @aschnell, does it really make sense in our case? Just curious.